plugins/pca/pca_output.py

```diff
@@ -76,12 +73,11 @@ def kernel_pca_output(pca: KernelPCA) -> (dict, int):
     kernel_pca_out["gamma"] = pca.gamma
     kernel_pca_out["degree"] = pca.degree
     kernel_pca_out["coef0"] = pca.coef0
-    dim = len(pca.eigenvectors_[0])
-    return kernel_pca_out, dim
+    return kernel_pca_out
 
 
 # dim = num features of output. Return dim here, since input params of the plugin can be <= 0
-def pca_to_output(pca: BaseEstimator) -> (dict, int):
+def pca_to_output(pca: BaseEstimator) -> dict:
```

Comment:
this could use <https://docs.python.org/3/library/functools.html#functools.singledispatch>